docs/module-guidelines.md

```diff
@@ -0,0 +1,184 @@
+# Module authoring guidelines
+
+Modules should adhere to following guidelines.
+
+## Terraform Requirements
+
+The module source field must point to a single terraform module. We recommend
+the following structure:
+
+* `main.tf` file composing the terraform resources using provided variables.
+* `variables.tf` file defining the variables used.
+* (Optional) `outputs.tf` file defining any exported outputs used (if any).
+* (Optional) `modules/` sub-directory pointing to submodules needed to create the
+  top level module.
+
+## General Best Practices
+
+* Variables for environment-specific values (like `project_id`) should not be
+  given defaults. This forces the calling module to provide meaningful values.
+* Variables should only have zero-value defaults (like null or empty strings)
+  where leaving the variable empty is a valid preference which will not be
+  rejected by the underlying API(s).
+* Set good defaults wherever possible. Be opinionated about HPC use cases.
+* Follow common variable [naming conventions](#use-common-names-and-types-for-common-variables).
+* If there are common hpc-toolkit variables already defined, then do not set defaults (`region`, `zone`, `project_id`, `deployment_name`, etc.)
+* All files should contain a license header. Headers can be added automatically
+  using `make add-google-license`.
```

Comment:
That only works if you add a google license.  Instead: headers can be added automatically using [addlicense](https://github.com/google/addlicense), or make add-google-license if adding a Google License.